### PR TITLE
Forest-map demonstrator

### PR DESCRIPTION
**Pull request by bbischke**
A new demonstrator for showcasing a large-scale forest type mapping with eo-learn and multi-temporal S2 imagery.

### CODE REVIEW

**Comment by batic** [comment]
Hi @bbischke 

I've tried running this and noticed some issues that have to be resolved. Before going into details, copying the data from the bucket doesn't work (`fatal error: An error occurred (403) when calling the HeadObject operation: Forbidden`)

Please re-upload the data to s3 using additional flag `--acl bucket-owner-full-control`; e.g., `aws s3 cp ./forest-map/ s3://queryplanet.sentinel-hub.com/forest-map/ --recursive --acl bucket-owner-full-control`

**Comment by bbischke** [comment]
Hello @batic, I re-uploaded the data to s3 with the additional flag

File: forest-map/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+A pre-trained model is open-sourced and available on the AWS S3 bucket [queryplanet.sentinel-hub.com](http://queryplanet.sentinel-hub.com/index.html). The results shown in the notebook are exemplary and are obtained with models fine-tuned on a subset of the dataset. The models are released under [CC BY-NC 4.0](https://creativecommons.org/licenses/by-nc/4.0/) license.
+
+
+
+## Requirements
+
+Before running the notebook, make sure to install the requirements with `pip install -r requirements.txt`.
+
+You need a Sentinel Hub account to run the example notebook, which you can get [here](https://services.sentinel-hub.com/oauth/subscription). Free trial accounts are available.
+
```
Comment (batic) [inline forest-map/README.md:32]: The readme would benefit from a "How to run" section, which would just be how to spin up `jupyter notebook` (which gets installed with reqs)

**Review by batic — COMMENTED** [review]
Some notes from trying out the notebook:

 * the `ScaleImageTask` changes the flow: if one runs the cell where `ScaleImageTask` is defined and then runs the next cell, the `workflow.execute` fails: `download_node` has been rewritten in the `ScaleImageTask` cell above. Suggestion: rename the nodes for `workflow_with_scale`
 * the area of interest is rather large, and some users might not have enough processing units left to run the whole area - please consider showing download and inference on a smaller subset (3x3?) of eo-patches, similar as has been done in [LULC eo-learn example](https://github.com/sentinel-hub/eo-learn/blob/master/examples/land-cover-map/SI_LULC_pipeline.ipynb)
 possible solution:
 ```python
 TILE_IDS = [33, 42, 51, 60,
            32, 41, 50, 59,
            31, 40, 49, 58,
            30, 39, 48, 57,]

## Run the workflow over selected eopatches
for idx in TILE_IDS:
    result = workflow.execute(
        {
            download_node : {'bbox': bbox_list[idx], 'time_interval': ['2021-07-01','2021-09-30']},
            save_node: {"eopatch_folder": f"eopatch_{idx}"},
        }
    )
```
 * if you do this (above), then make sure the visualization cell works (e.g. using `eo-patch_0` and get `IndexError: index 2 is out of bounds for axis 0 with size 2`)

 * if user doesn't have GPU, loading of the model fails; it would be great if how-to would be added, or even code changed so that user selects what they have available. One possible way would be to define `device` when initing the `InferenceTask`, because if using cpu, `load_model` should have additional parameter (`map_location=torch.device('cpu')`)

Some suggestions to make the notebook nicer to read (feel free to ignore):

```python
with fiona.open(geo_file, layer="DE044L1_KAISERSLAUTERN_UA2018_Boundary") as layer:
    feature = list(layer.values())[0] 
    aoi_shape = shape(feature['geometry'])
    aoi_crs = layer.crs
```

could be replaced with

```python
aoi_gdf = gpd.read_file(geo_file, layer="DE044L1_KAISERSLAUTERN_UA2018_Boundary")
aoi = aoi_gdf.geometry[0]
aoi_crs = aoi_gdf.crs
```

 `bbox_splitter` would then become:

```python
bbox_splitter = UtmZoneSplitter([aoi], aoi_crs, 5000)
```

and `bbox_gdf`:

```python
bbox_splitter = UtmZoneSplitter([aoi], aoi_crs, 5000)

bbox_list = bbox_splitter.get_bbox_list()
info_list = bbox_splitter.get_info_list()

# Prepare info of selected EOPatches
geometry = [bbox.geometry for bbox in bbox_list]

bbox_gdf = gpd.GeoDataFrame(data=info_list, geometry=geometry, crs=a.crs).to_crs("EPSG:3035")

# Save to shapefile
bbox_gdf.to_file(os.path.join(".", "grid_kl_500x500.gpkg"), driver="GPKG")
```

File: forest-map/README.md [hunk 2]
```diff
@@ -13,23 +13,25 @@ The code in the `forest-map.ipynb` notebook allows executing forest segmentation
...
 
 ## Pre-trained model
 
 A pre-trained model is open-sourced and available on the AWS S3 bucket [queryplanet.sentinel-hub.com](http://queryplanet.sentinel-hub.com/index.html). The results shown in the notebook are exemplary and are obtained with models fine-tuned on a subset of the dataset. The models are released under [CC BY-NC 4.0](https://creativecommons.org/licenses/by-nc/4.0/) license.
 
-
-
 ## Requirements
 
 Before running the notebook, make sure to install the requirements with `pip install -r requirements.txt`.
```
Comment (batic) [inline forest-map/README.md:27]: One last thing to add to requirements: `awscli`. Without it, users can't list or download data from the bucket.

**Comment by batic** [comment]
Hi @bbischke ,

After you add `awscli` to the requirements, I will merge your PR. I've run the notebook, nice work!

Regards,
Matej

**Comment by bbischke** [comment]
Hey @batic,

I added the `awscli` package to the requirements.txt. 

Thank you again for your great feedback above! 

Best regards,
Benjamin

**Pull request merged**